x/wasm/ibc.go

```diff
@@ -19,13 +17,20 @@ import (
 
 var _ porttypes.IBCModule = IBCHandler{}
 
+// internal interface that is implemented by ibc middleware
```

Comment:
I assume this works regardless of the stack? (eg no middleware or with middleware)

This looks much cleaner than the previous approach.